src/sa_web/static/js/views/map-view.js

```diff
@@ -114,10 +130,12 @@ var Shareabouts = Shareabouts || {};
       });
 
       // Bind data events
-      self.collection.on('reset', self.render, self);
-      self.collection.on('add', self.addLayerView, self);
-      self.collection.on('remove', self.removeLayerView, self);
-
+      _.each(self.collection, function(collection) {
+        collection.on('reset', self.render, self);
+        collection.on('add', self.addLayerView, self);
+        collection.on('remove', self.removeLayerView, self);
+      }); 
+      
```

Comment:
Yep, this extra loop makes sense, although we may want to update `addLayerView` and `removeLayerView` to have them accept a collection's dataset id (or the collection itself), and return a closure that accepts a model. You can view the `addLandmarkLayerView` and `removeLandmarkLayerView` methods below for guidance. Ideally, we could refactor the "LandmarkLayerView" and "LayerView" methods into a single method, but it's not a priority.